### ISSUE TITLE
yyinput() with EOF has inconsistent behavior with documentation

### ISSUE THREAD

**Issue by forewing** [westes/flex#444]
# Current Behavior

As documentation showed, `yyinput()` return EOF when encountering an end-of-file:

https://github.com/westes/flex/blob/8b1fbf674f2e038df9cf1fe7725617e3837ae2a9/doc/flex.texi#L4549-L4553

And the example of comments discarding use EOF to check:

https://github.com/westes/flex/blob/8b1fbf674f2e038df9cf1fe7725617e3837ae2a9/doc/flex.texi#L1410-L1411

https://github.com/westes/flex/blob/8b1fbf674f2e038df9cf1fe7725617e3837ae2a9/doc/flex.texi#L1422

But f863c9490e6912ffcaeb12965fb3a567a10745ff changed the return value of `yyinput()` from `EOF` to `0` when encountering an end-of-file:

https://github.com/westes/flex/blob/3a4af3424c3df6a21eda0c77dd6e8ee4a46c997e/src/flex.skl#L1970-L1973

https://github.com/westes/flex/blob/f863c9490e6912ffcaeb12965fb3a567a10745ff/src/flex.skl#L1970-L1973

# Possible Solution

The documentation should be updated.

**Comment by zmajeed** [comment]
Could you provide a reproducible test case that shows yyinput() does not return EOF on end of file?

Isn't the code change you've pointed to what's described in chapter 9 of the doc?

"When the scanner receives an end-of-file indication from YY INPUT, it then checks the yywrap() function. If yywrap() returns false (zero), then it is assumed that the function has gone ahead and set up yyin to point to another input file, and scanning continues. If it returns true (non-zero), then the scanner terminates, returning 0 to its caller."

**Comment by zmajeed** [comment]
You're right - `input()` is shown returning EOF on end of file in some parts of flex docs while it returns 0 in other places - I found the Posix spec for `input()` - https://pubs.opengroup.org/onlinepubs/9699919799/utilities/lex.html - it's supposed to return 0 not EOF on end of file

int  input(void)

Returns the next character from the input, or zero on end-of-file. It shall obtain input from the stream pointer yyin, although possibly via an intermediate buffer. Thus, once scanning has begun, the effect of altering the value of yyin is undefined. The character read shall be removed from the input stream of the scanner without any processing by the scanner.
